geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySenderEventProcessor.java

```diff
@@ -1225,6 +1228,13 @@ public void closeProcessor() {
     }
   }
 
+  protected GatewaySenderEventImpl.TransactionMetadataDisposition getTransactionMetadataDisposition(
+      final boolean isLastEventInTransaction) {
+    return getSender().mustGroupTransactionEvents()
```

Comment:
I think this would be easier to understand if you changed it to use if statements.